reference/src/glossary.md

```diff
@@ -1,5 +1,49 @@
 ## Glossary
 
+#### Aliasing
+
+(Please note: a full aliasing model for Rust has not yet been constructed, but
+at the moment we can give the following definition.)
+
+*Aliasing* is any time one pointer or reference points to a "span" of memory
+that overlaps with the span of another pointer or reference. A span of memory is
+similar to how a slice works: there's a base byte address as well as a length in
+bytes.
+
+Consider the following example:
+
+```rust
+fn main() {
+    let u: u64 = 7_u64;
+    let r: &u64 = &u;
+    let s: &[u8] = unsafe {
+        core::slice::from_raw_parts(&u as *const u64 as *const u8, 8)
+    };
+    let (head, tail) = s.split_first().unwrap();
+}
+```
+
+In this case, both `r` and `s` alias each other, since they both point to all of
+the bytes of `u`.
+
+However, `head` and `tail` do not alias each other: `head` points to the first
+byte of `u` and `tail` points to the other seven bytes of `u` after it. Also,
+both `head` and `tail` alias `s`.
+
+* The span length of `&T`, `&mut T`, `*const T`, or `*mut T` when `T` is
```

Comment:
What about adding a header, to bind `t`? "The span length of a pointer or reference `t` is defined as follows:"

Reply:
yeah lemme try something like that

Reply:
Alright, I think I improved that part quite a bit.